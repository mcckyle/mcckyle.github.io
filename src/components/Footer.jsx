//Filename: Footer.jsx
//Author: Kyle McColgan
//Date: 31 August 2025
//Description: This file contains the React component for the website footer section.

import React from 'react';

function Footer()
{
    return (
        <footer className = "site-footer">
            <p className = "footer-text">
                Welcoming the calm of September üçÇ | Designed by <strong>Kyle McColgan</strong>, 2025.
            </p>
        </footer>
    );
}

export default Footer;
