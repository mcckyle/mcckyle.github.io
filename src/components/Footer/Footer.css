/* Filename: Footer.css */

/* =======================
   Footer Styles.
========================== */
.footer
{
    background: linear-gradient(
        180deg,
        color-mix(in oklab, var(--footer-bg), white 12%) 0%,
        color-mix(in oklab, var(--accent), var(--footer-bg) 50%) 100%
    );
    padding: 2rem 1rem;
    border-top: 1px solid var(--rule);
}

.footer-container
{
    margin: 0 auto;
    max-width: 800px;
    text-align: center;
    display: flex;
    flex-direction: column;
    gap: 0.5rem;
}

.footer-seasonal
{
    font-size: 1rem;
    font-weight: 500;
    color: var(--text);
}

.footer-credit
{
    font-size: 0.85rem;
    color: var(--text-muted);

    transition: color var(--transition);
}

.footer-credit strong
{
    font-weight: 600;
    color: var(--accent);
}

.footer-credit:hover
{
    color: var(--accent);
}
