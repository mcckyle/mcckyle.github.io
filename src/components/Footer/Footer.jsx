//Filename: Footer.jsx
//Author: Kyle McColgan
//Date: 29 September 2025
//Description: This file contains the component for the personal React project footer section.

import React from 'react';
import './Footer.css';

function Footer()
{
    return (
        <footer className="footer" aria-label="Footer">
          <div className="footer-container">
            <p className="footer-seasonal">Welcoming the calm of September üçÇ </p>
            <p className="footer-credit">
              Designed by <strong>Kyle McColgan</strong>, 2025.
            </p>
          </div>
        </footer>
    );
}

export default Footer;
