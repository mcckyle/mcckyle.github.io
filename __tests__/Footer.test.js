//Filename: Footer.test.js
//Author: Kyle McColgan
//Date: 29 September 2025
//Description: This file contains the Jest unit tests for the personal React project footer component.

import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Footer from '../src/components/Footer/Footer.jsx';

describe('Footer Component', () => {
  beforeEach(() => {
    render(<Footer />);
  });

  //Test #1
  // test('renders without crashing', () => {
  //   render(<Footer />);
  //   // No assertion necessary; render should not be thrown...
  // });

  //Test #2
  test('renders a footer element with the correct role.', () => {
    const footer = screen.getByRole('contentinfo');
    expect(footer).toBeInTheDocument();
    expect(footer.tagName.toLowerCase()).toBe('footer');
  });

  //Test #3
  test('footer contains a paragraph with the correct class.', () => {
    const paragraph = screen.getByText(/Welcoming/i);
    expect(paragraph).toHaveClass('footer-seasonal');
  });

  //Test #4
  test('footer displays the current year dynamically.', () => {
    const currentYear = new Date().getFullYear().toString();
    expect(screen.getByText(new RegExp(currentYear))).toBeInTheDocument();
  });

  //Test #5
  test('text content includes the word "September" and an emoji.', () => {
    const text = screen.getByText(/September/i);
    expect(text.textContent).toMatch(/🍂/);
  });

  //Test #6
  // test('footer text includes the © symbol.', () => {
  //   render(<Footer />);
  //   const text = screen.getByText(/©/);
  //   expect(text).toBeInTheDocument();
  // });

  //Test #7
  test('footer only contains one child element (the paragraph).', () => {
    const footer = screen.getByRole('contentinfo');
    expect(footer.childElementCount).toBe(1);
  });

  //Test #8
  test('text content includes a date', () => {
    const text = screen.getByText(/2025/);
    expect(text).toBeInTheDocument();
  });

  //Test #9
  test('footer has the correct class for styling.', () => {
    const footer = screen.getByRole('contentinfo');
    expect(footer).toHaveClass('footer');
  });

  //Test #10
  test('footer matches the snapshot.', () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });
});
