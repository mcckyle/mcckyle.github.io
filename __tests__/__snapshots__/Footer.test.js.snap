// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component footer matches the snapshot. 1`] = `
<DocumentFragment>
  <footer
    aria-label="Footer"
    class="footer"
  >
    <div
      class="footer-container"
    >
      <p
        class="footer-seasonal"
      >
        Welcoming the calm of September üçÇ 
      </p>
      <p
        class="footer-credit"
      >
        Designed by 
        <strong>
          Kyle McColgan
        </strong>
        , 2025.
      </p>
    </div>
  </footer>
</DocumentFragment>
`;
